### PR TITLE
Added CheckNoDataAgent

### PR DESCRIPTION
**Pull request by tharmaman**

### CODE REVIEW

File: src/foam/u2/crunch/wizardflow/CheckNoDataAgent.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    async function execute() {
+      var shouldOpen = false;
+
+      await this.crunchService.getGrantPath(null, this.rootCapability.id).then(capabilitiesNeeded => {
+        capabilitiesNeeded.forEach(capa => {
+          if ( capa.of ){
+            shouldOpen = true;
+          }
+
+          if ( ! shouldOpen ) {
```
Comment (KernelDeimos) [inline src/foam/u2/crunch/wizardflow/CheckNoDataAgent.js]: This looks like it should be after the forEach

**Review by KernelDeimos — CHANGES_REQUESTED** [review]

File: src/foam/u2/crunch/wizardflow/CheckNoDataAgent.js (hunk 1 above)
Comment (tharmaman) [reply src/foam/u2/crunch/wizardflow/CheckNoDataAgent.js]: true

File: src/foam/u2/crunch/wizardflow/CheckNoDataAgent.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      name: 'submitted'
+    }
+  ],
+
+  methods: [
+    // If Property expressions ever unwrap promises this method can be blank.
+    async function execute() {
+      var shouldOpen = false;
+
+      await this.crunchService.getGrantPath(null, this.rootCapability.id).then(capabilitiesNeeded => {
```
Comment (KernelDeimos) [inline src/foam/u2/crunch/wizardflow/CheckNoDataAgent.js]: can do `capabilitiesNeeded = ...` because `await`

**Review by KernelDeimos — APPROVED** [review]

File: src/foam/u2/crunch/wizardflow/CheckNoDataAgent.js (hunk 2 above)
Comment (tharmaman) [reply src/foam/u2/crunch/wizardflow/CheckNoDataAgent.js]: Ya refactored

**Pull request merged**